src/main/java/com/jakewharton/retrofit2/converter/kotlinx/serialization/Serializer.kt

```diff
@@ -2,41 +2,36 @@ package com.jakewharton.retrofit2.converter.kotlinx.serialization
 
 import kotlinx.serialization.KSerialLoader
 import kotlinx.serialization.KSerialSaver
+import kotlinx.serialization.json.JSON
+import kotlinx.serialization.protobuf.ProtoBuf
 import okhttp3.MediaType
 import okhttp3.RequestBody
 import okhttp3.ResponseBody
 
-@Suppress("UNCHECKED_CAST") // Widening T to Any
 internal sealed class Serializer {
-  abstract fun <T> fromResponseBody(loader: KSerialLoader<T>, body: ResponseBody): T
-  abstract fun <T> toRequestBody(contentType: MediaType, saver: KSerialSaver<T>, value: T): RequestBody
+  abstract fun <T : Any> fromResponseBody(loader: KSerialLoader<T>, body: ResponseBody): T
```

Comment:
the type is already cast to a non-null type below, anyway.
Retrofit doesn't hand off the type nullability information, so the KSerialLoaders/Savers here use non-null types and won't handle null like: JSON `"null"`.